internal/engine/build_and_deployer.go

```diff
@@ -71,13 +71,20 @@ func (composite *CompositeBuildAndDeployer) BuildAndDeploy(ctx context.Context,
 		if _, ok := err.(RedirectToNextBuilder); ok {
 			logger.Get(ctx).Debugf("(expected error) falling back to next build and deploy method "+
 				"after error: %v", err)
-		} else if i+1 < len(composite.builders) {
-			logger.Get(ctx).Infof("falling back to next build and deploy method "+
-				"after unexpected error: %v", err)
+		} else {
+			lastUnexpectedErr = err
+			if i+1 < len(composite.builders) {
+				logger.Get(ctx).Infof("falling back to next build and deploy method "+
```

Comment:
this still seems like a confusing thing to print in the docker-compose case, where there are more deployers but none of them are applicable

Reply:
cool, i'll change it to something that surfaces the error w/o necessarily indicating that we're falling back.